idaes/power_generation/unit_models/boiler_heat_exchanger_2D.py

```diff
@@ -2352,6 +2502,21 @@ def initialize(blk, shell_state_args=None, tube_state_args=None,
         blk.heat_tube_eqn.deactivate()
         blk.heat_shell_eqn.deactivate()
 
+        if blk.config.has_header is True:
+            blk.head_heat_conduction_eqn.deactivate()
```

Comment:
Minor comment: It might be work making this a list of names that you can iterate over to activate and deactivate constraints. This will save repeating all these names again later when you reactivate them, and make it less likely that something will accidentally be forgotten.